### PR TITLE
Feature/prepare cmake workflow presets

### PR DESCRIPTION
**Pull request by ClausKlein**

### CODE REVIEW

**Review by dietmarkuehl — CHANGES_REQUESTED** [review]
Overall these changes are a great contribution: Thank you very much!

I'm certainly not a `cmake` expert and I don't fully understand most of the changes. I understand the objective and the result seems to work. My main discomfort with this change is that after this change the project requires `ninja` to build (before it could build with `ninja` or `make`; I appreciate that the `Makefile` got updated while the recommended build is meant to use `cmake`). Maybe it is time to accept the requirement for `ninja`, though.

There are two things which I think do need changing:

1. There is a code change to return `get_domain` instead of `default_domain{}`. That is incorrect and the proper fix is to include the correct header (also commented on).
2. The `find-package-test` takes rather long. I don't know if it can be made [much] faster. Thus, I think there needs to be a way to somehow disable the test for development.

Finally, there is something I'm not quite happy with: When trying to compile as is, I do get `cmake` errors about `cmake` not knowing how to enable C++26:

```
  Target "cmTC_ba922" requires the language dialect "CXX26" .  But the
  current compiler "GNU" does not support this, or CMake does not know the
  flags to enable it.
```

This error is a bit odd considering that I can build the `main` branch of the repository enabling this same version setting the C++ version to 26 without any problems. The `cmake` version installed is 3.30.3.

Using C++ version 23 or upgrading `cmake` to version 3.30.5 fixes this problem. However, I don't have `cmake` 3.30.5 available on the Raspberry PI distribution and it seems a bit hostile to insist on the very latest version of `cmake`. Something seems odd over here.

The current implementation doesn't require C++26 to build - it can build with C++23. It would use some C++26 features if it detects them to be present based on feature test macros and it would be nice to verify that this code does work but that may not be essential. Thus, it is probably reasonable to set the default to use C++23. When I did that I needed to change both `CMakeLists.txt` and `CMakePresets.json`. Ideally, only one place would need to get changed.

File: src/beman/execution26/tests/CMakeLists.txt [hunk 1]
```diff
@@ -83,11 +86,47 @@ list(APPEND execution_tests
...
+        set(CMAKE_BUILD_TYPE Release)
+    endif()
+    add_test(find-package-test
+        ${CMAKE_CTEST_COMMAND}
+        --verbose
+        # TODO: --output-on-failure
+        -C ${CMAKE_BUILD_TYPE}
+        --build-and-test
+            "${CMAKE_CURRENT_SOURCE_DIR}"
+            "${CMAKE_CURRENT_BINARY_DIR}/find-package-test"
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply src/beman/execution26/tests/CMakeLists.txt:118]: changed, I use only one test in this case.

File: include/beman/execution26/detail/sched_env.hpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ namespace beman::execution26::detail
             if constexpr (requires{ this-> sched.query(q); })
                 return this->sched.query(q);
             else
-                return ::beman::execution26::default_domain{};
+                return ::beman::execution26::get_domain;
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply include/beman/execution26/detail/sched_env.hpp]: done

**Review by dietmarkuehl — APPROVED** [review]

**Comment by ClausKlein** [comment]
> There are two things which I think do need changing:
> 
> 1. There is a code change to return `get_domain` instead of `default_domain{}`. That is incorrect and the proper fix is to include the correct header (also commented on).

done

> 2. The `find-package-test` takes rather long. I don't know if it can be made [much] faster. Thus, I think there needs to be a way to somehow disable the test for development.

done

> Using C++ version 23 or upgrading `cmake` to version 3.30.5 fixes this problem. However, I don't have `cmake` 3.30.5 available on the Raspberry PI distribution and it seems a bit hostile to insist on the very latest version of `cmake`. Something seems odd over here.

try to use: `pip install cmake ninja` to get the newest version available.

> 
> The current implementation doesn't require C++26 to build - it can build with C++23. It would use some C++26 features if it detects them to be present based on feature test macros and it would be nice to verify that this code does work but that may not be essential. Thus, it is probably reasonable to set the default to use C++23. When I did that I needed to change both `CMakeLists.txt` and `CMakePresets.json`. Ideally, only one place would need to get changed.

I will have a try to fix it ...

**Comment by dietmarkuehl** [comment]
Looks good! Thank you!

**Comment by ClausKlein** [comment]
I found a solution for the c++ std issue, see https://github.com/ClausKlein/execution26/pull/1/commits/bc5253464eec158cbb68b3e8cf16a21d0906ffe8

P.S: what do you mean about [clang-format](https://cmake-format.readthedocs.io/en/latest/index.html)?

**Comment by dietmarkuehl** [comment]
Re C++26: thanks for looking into it. How does that interact with the project becoming a subproject of something else, setting whatever choice is set there? I _think_ the language choice from the top level project should be used. At the moment the only plausible choices are C++23 and C++26. I can imagine that much of `std::execution` can be made to work with C++20 (it is one of the things I want to look into; earlier revisions won't work due to the use of concepts). Soon C++29 should become an option (and I wonder if that will require another `cmake` change).

re `clang-format` (or `clang-tidy` for that matter): it should be used! I just haven't come around setting any of that up. If there is a setup I can run it manually on all the code to get things fixed - I imagine the resulting PR would be quite big but if I run it and tests pass I'd be OK to pretty much just landing it. I think [beman-exemplar](https://github.com/beman-project/exemplar) has a [configuration](https://github.com/beman-project/exemplar/blob/main/.clang-format) for `clang-format`.

When building yesterday with your setup I found that there were some warnings: I hope to fix them really soon as well (although probably not today as I have a major chunk of work to do today - write wording for `simple_counting_scope` based on the description in http://wg21.link/p3149 ).

I haven't run `clang-tidy` directly, yet (at work it gets run by our CI). I only found yesterday that I'll need a configuration to be meaningful. I'd also want to clean reports from that up, at least for reasonable configurations: apparently some of the reports are likely false positives.

**Pull request merged**